subsys/dfu/dfu_target/src/dfu_target_full_modem.c

```diff
@@ -70,7 +75,8 @@ int dfu_target_full_modem_done(bool successful)
 
 	if (successful) {
 		LOG_INF("Modem firmware downloaded to flash device");
+		return stream_flash_finish(&stream, true);
```

Comment:
Maybe not at this level of detail yet, but why is this now only done on success?

Reply:
stream_flash_finish() should be equivalent to the flush write + clear settings that is done in the original dfu_target_stream_done() on success. I forgot the failure branch here though. I think this code reads a bit weird too, so I will have a look at the API/naming.